classes/ObjectModel.php

```diff
@@ -686,12 +694,37 @@ public function duplicateObject()
      *
      * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
+     * @throws PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException
      */
     public function update($null_values = false)
+    {
+        $currentClassName = $this->getObjectName();
+        $oldObject = new $currentClassName($this->id, $this->id_lang, $this->id_shop);
+        $comparator = new ObjectModelComparator($oldObject, $this);
+
+        if ($this->skipUpdateIfUnchanged && !$comparator->hasChanges()) {
+            return true;
+        }
+
+        return $this->processUpdateAction((bool) $null_values, $comparator);
+    }
+
+    /**
+     * Updates the current object in the database.
+     *
+     * @param bool $null_values
+     * @param ObjectModelComparator|null $comparator
+     *
+     * @return bool
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    protected function processUpdateAction(bool $null_values = false, ?ObjectModelComparator $comparator = null): bool
     {
         // @hook actionObject<ObjectClassName>UpdateBefore
-        Hook::exec('actionObjectUpdateBefore', ['object' => $this]);
-        Hook::exec('actionObject' . $this->getFullyQualifiedName() . 'UpdateBefore', ['object' => $this]);
+        Hook::exec('actionObjectUpdateBefore', ['object' => $this, 'objectComparator' => $comparator]);
```

Comment:
I do not think the `objectComparator` should be passed.

Reply:
humm why @mvorisek ?